File: app/src/main/java/com/werocksta/dagger2demo/di/component/ActivityComponent.java
package com.werocksta.dagger2demo.di.component;

import com.werocksta.dagger2demo.di.ActivityScope;
import com.werocksta.dagger2demo.di.module.ActivityModule;
import com.werocksta.dagger2demo.view.fragment.MainFragment;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

}

File: app/src/main/java/com/werocksta/dagger2demo/di/component/ActivityComponent.kt
package com.werocksta.dagger2demo.di.component

import com.werocksta.dagger2demo.di.PerActivity
import com.werocksta.dagger2demo.di.module.ActivityModule

import dagger.Component

@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent
